File: py/643.py
class Solution:
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        s = 0
        for i in nums[0:k-1]:
            s += i
        m = -10000000
        for i in range(k-1, len(nums)):
            s += nums[i]
            r = s / k
            if m < r:
                m = r
            # print(i, nums[i], s, r)
            s -= nums[i-k+1]
        return m


s = Solution()
print(s.findMaxAverage([1, 12, -5, -6, 50, 3], 4))

File: js/643.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {

  let sum = 0
  for (let i = 0; i < k - 1; i++) {
    sum += nums[i]
  }
  let max = -Math.pow(2, 31)
  for (let i = k - 1; i < nums.length; i++) {
    sum += nums[i]
    const res = sum / k
    if (max < res) {
      max = res
    }
    console.log({ i, num: nums[i], sum, res });
    
    sum -= nums[i - k + 1]
  }
  return max
};

var assert = require('assert');
var nums = [1, 12, -5, -6, 50, 3],
  k = 4
assert.equal(findMaxAverage(nums, k), 12.75)